File: OccultMerchant/OccultMerchant/items/Price.cs
namespace OccultMerchant.items
{
    public enum CoinType
    {
        PlatinumCoin,
        GoldCoin,
        SilverCoin,
        CopperCoin
    }
    
    public struct Price
    {
        public int value { get; set; }
        public CoinType coin { get; set; }

       public Price(int _value, CoinType _coin)
        {
            this.value = _value;
            this.coin = _coin;
        }
    }

    
}

File: Showcase/src/app/Items/Price.ts

export enum CoinType {
  PlatinumCoin=3,
  GoldCoin=2,
  SilverCoin=1,
  CopperCoin=0
}


export class Price {
  public value: number;
  public coin: CoinType;

  constructor(_value: number = 0, _coin: CoinType = CoinType.SilverCoin) {
    this.value = _value;
    this.coin = _coin;
  }

  toString(): string{
    return this.value +" " + this.coin;
  }

  public clone(): any {
    var cloneObj = new Price() ;
    for (var attribut in this) {
      if (typeof this[attribut] === "object") {
        // @ts-ignore
        cloneObj[attribut] = this[attribut].clone();
      } else {
        // @ts-ignore
        cloneObj[attribut] = this[attribut];
      }
    }
    return cloneObj;
  }
}
